### ISSUE TITLE
Feature request: Support for linked files

### ISSUE THREAD

**Issue by jnyrup** [stryker-mutator/stryker-net#523]
**Is your feature request related to a problem? Please describe.**
My project uses `<Compile Include="..\MyAttributes.cs" Link="MyAttributes.cs" />` to include a file outside the project folder in the compilation.

When running Stryker 0.11.0 I get an error, that `MyAttributes.cs` cannot be found.

```
PS C:\Users\jonas\source\repos\StrykerCompileLink\StrykerTests> dotnet stryker --log-file

   _____ _              _               _   _ ______ _______
  / ____| |            | |             | \ | |  ____|__   __|
 | (___ | |_ _ __ _   _| | _____ _ __  |  \| | |__     | |
  \___ \| __| '__| | | | |/ / _ \ '__| | . ` |  __|    | |
  ____) | |_| |  | |_| |   <  __/ |    | |\  | |____   | |
 |_____/ \__|_|   \__, |_|\_\___|_| (_)|_| \_|______|  |_|
                   __/ |
                  |___/


Version 0.11.0 (beta)

[20:44:15 INF] Using C:\Users\jonas\source\repos\StrykerCompileLink\StrykerTests\StrykerTests.csproj as project file
[20:44:19 INF] Started initial build using dotnet build
[20:44:21 INF] Initial build successful
[20:44:21 INF] Using testrunner DotnetTest
[20:44:21 INF] Initial testrun started
[20:44:24 INF] Total number of tests found in initial test run: 1
[20:44:24 INF] Initial testrun successful in 2270 ms
[20:44:24 INF] Using 33405 ms as testrun timeout
[20:44:26 WRN] Stryker.NET encountered an compile error in C:\Users\jonas\source\repos\StrykerCompileLink\StrykerCompileLink\Math.cs with message: The type or namespace name 'MyAttributes' could not be found (are you missing a using directive or an assembly reference?) (Source code: MyAttributes)
[20:44:26 WRN] Safe Mode! Stryker will try to continue by rolling back all mutations in method. This should not happen, please report this as an issue on github with the previous error message.
[20:44:26 FTL] Stryker.NET could not compile the project after mutation. This is probably an error for Stryker.NET and not your project. Please report this issue on github with the previous error message.
[20:44:26 ERR] An error occurred during the mutation test run
System.ApplicationException: Internal error due to compile error.
   at Stryker.Core.Compiling.RollbackProcess.RemoveMutantIfStatements(SyntaxTree originalTree, ICollection`1 diagnosticInfo, Boolean devMode) in D:\a\1\s\src\Stryker.Core\Stryker.Core\Compiling\RollbackProcess.cs:line 188
   at Stryker.Core.Compiling.RollbackProcess.Start(CSharpCompilation compiler, ImmutableArray`1 diagnostics, Boolean devMode) in D:\a\1\s\src\Stryker.Core\Stryker.Core\Compiling\RollbackProcess.cs:line 61
   at Stryker.Core.Compiling.CompilingProcess.TryCompilation(MemoryStream ms, CSharpCompilation compilation, EmitResult previousEmitResult, Boolean devMode, Int32 retryCount) in D:\a\1\s\src\Stryker.Core\Stryker.Core\Compiling\CompilingProcess.cs:line 108
   at Stryker.Core.Compiling.CompilingProcess.Compile(IEnumerable`1 syntaxTrees, MemoryStream ms, Boolean devMode) in D:\a\1\s\src\Stryker.Core\Stryker.Core\Compiling\CompilingProcess.cs:line 76
   at Stryker.Core.MutationTest.MutationTestProcess.Mutate(StrykerOptions options) in D:\a\1\s\src\Stryker.Core\Stryker.Core\MutationTest\MutationTestProcess.cs:line 91
   at Stryker.Core.StrykerRunner.RunMutationTest(StrykerOptions options) in D:\a\1\s\src\Stryker.Core\Stryker.Core\StrykerRunner.cs:line 77
[20:44:26 INF] Time Elapsed 00:00:10.6122125
```

**Describe the solution you'd like**
The project should compile.

**Describe alternatives you've considered**

**Additional context**
Full log: The compilation crashes before producing a log
Full example: 
[StrykerCompileLink.zip](https://github.com/stryker-mutator/stryker-net/files/3199368/StrykerCompileLink.zip)

**Comment by richardwerkman** [comment]
Currently stryker indeed only scans for `*.cs` files in the project directory. We don't check for external references.

We could look into buildalyzers ability to find all files inside a project. If it also finds linked files we should switch from scanning the file system to buildalyzers output. This could also potentionally fix #128.

If buildalyzer also fails to find linked files (or nuget content files) we better keep the scanning for content files to ourselves. Then we can look for `compile` and `include` in the project file.

@Mobrockers agreed?

**Comment by Mobrockers** [comment]
@richardwerkman I feel confident buildalyzer is sufficiently mature and under development that we can use it to find our project sources. If this changes in the future we can simply restore our own file-finding at that point.

That is assuming buildalyzer can already find more files than we currently can, if not then we'll have to build it ourselves anyway so then I would not like to use buildalyzer for it.

**Comment by Mobrockers** [comment]
Unfortunately it seems like buildalyzer will not be able to help us with this, but I have an implementation for this feature in the works

**Comment by dupdob** [comment]
It looks like this issue should be fixed since V 0.16

**Comment by jnyrup** [comment]
I can confirm that this issue has been resolved.
